src/Marten.AsyncDaemon.Testing/cross_stream_aggregation.cs

```diff
@@ -0,0 +1,152 @@
+    {
+        new CrossStreamDayProjection().Lifecycle.ShouldBe(ProjectionLifecycle.Async);
+    }
+
+    [Fact]
+    public async Task splicing_events()
+    {
+        NumberOfStreams = 10;
+        await PublishMultiThreaded(3);
+
+        var allEvents = await theSession.Events.QueryAllRawEvents().ToListAsync();
+
+        var slicer = new CrossStreamDayProjection().As<IEventSlicer<Day, int>>();
+
+        var slices = await slicer.SliceAsyncEvents(theSession, allEvents.ToList());
+
+        foreach (var slice in slices.SelectMany(x => x.Slices).ToArray())
+        {
+            var events = slice.Events();
+            events.All(x => x.Data is IDayEvent || x.Data is Movement).ShouldBeTrue();
+            events.Select(x => x.Data).OfType<IDayEvent>().All(x => x.Day == slice.Id)
+                .ShouldBeTrue();
+
+            var travels = events.OfType<Event<Travel>>().ToArray();
+            foreach (var travel in travels)
+            {
+                var index = events.As<List<IEvent>>().IndexOf(travel);
+
+                for (var i = 0; i < travel.Data.Movements.Count; i++)
+                {
+                    events.ElementAt(index + i + 1).Data.ShouldBe(travel.Data.Movements[i]);
+                }
+            }
+        }
+
+        slices.ShouldNotBeNull();
+    }
+
+    [Fact]
+    public async Task run_end_to_end()
+    {
+        StoreOptions(x => x.Projections.Add(new CrossStreamDayProjection()));
+
+        await theStore.EnsureStorageExistsAsync(typeof(Day));
+
+        using var agent = await StartDaemon();
+
+        NumberOfStreams = 10;
+        await PublishSingleThreaded();
+
+        _output.WriteLine($"Expecting {NumberOfEvents} events");
+
+        await agent.Tracker.WaitForShardState("Day:All", NumberOfEvents, 2.Minutes());
+
+        var days = await theSession.Query<Day>().ToListAsync();
+
+        var allEvents = await theSession.Events.QueryAllRawEvents().ToListAsync();
+        var dayEvents = allEvents.Select(x => x.Data).OfType<IDayEvent>();
+        var groups = dayEvents.GroupBy(x => x.Day).ToList();
+
+        foreach (var day in days)
+        {
+            var matching = groups.FirstOrDefault(x => x.Key == day.Id);
+            matching.ShouldNotBeNull();
+
+            day.Started.ShouldBe(matching.OfType<TripStarted>().Count());
+            day.Ended.ShouldBe(matching.OfType<TripEnded>().Count());
+            day.East.ShouldBe(matching
+                .OfType<Travel>()
+                .SelectMany(x => x.Movements)
+                .Where(x => x.Direction == Direction.East)
+                .Sum(x => x.Distance));
+        }
+    }
+}
+
+public class CrossStreamDayProjection: CrossStreamAggregateProjection<Day, int>
+{
+    public CrossStreamDayProjection()
+    {
+        ProjectionName = "Day";
+
+        // You have to specify this in order for the Travel events
+        // to flow through this projection.
+        IncludeType<Travel>();
```

Comment:
**SUGGESTION:** I think that it'd be better to rename `IncludeType` to `IncludeEventType` or just `Include` as I had a hard time remembering what it does. Also, we should rename `Travel` to something that's named more like an event, e.g. `TravelFinished`, as people will have a hard time understanding what it does. My first thought is that's some new feature to include other documents.